### PR TITLE
move lcp from image list

### PR DESCRIPTION
**Pull request by yaniv-li**
I had to re-implement the lcp:
1. make sure it gets analyzed and not filtered out 
2. return the analyzed object and not the speed test result 
3. remove the lcp from the general image list so we would not have duplication.

### CODE REVIEW

File: cloudinary/apiCaller.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const cloudinaryParser = require('./cloudinaryResultParser');
 const cloudinary = require('cloudinary');
 const async = require('async');
 const request = require('request');
+const path = require('path');
```
Comment (mikeys) [inline cloudinary/apiCaller.js:15]: Where is this used?

File: cloudinary/apiCaller.js [hunk 2]
```diff
@@ -73,6 +78,9 @@ const sendToCloudinery = (imagesArray, batchSize, dpr, metaData, quality, cb, ro
       cb(parsed, null);
       return;
     }
+    // move lcp
+    const lcpIdx = parsed.imagesTestResults.findIndex((i) => i.tags.includes('lcp'));
+    metaData.lcp = parsed.imagesTestResults.splice(lcpIdx, 1)[0];
```
Comment (mikeys) [inline cloudinary/apiCaller.js:83]: What if no Lcp was found?

File: wtp/wtpResultsParser.js [hunk 3]
```diff
@@ -61,6 +58,15 @@ const parseTestResults = (testJson) => {
         userAgent = head.split(':').pop();
       }
     });
+    // make sure LCP image is in image list to analyze
+    const inList = imageList.findIndex((image) => {
+      return image.url === lcpURL;
+    });
+    if (inList === -1) {
+      imageList.push(origImageList.find((image) => {
```
Comment (mikeys) [inline wtp/wtpResultsParser.js:66]: Wouldn’t normally origImageList will contain the imageList first? In that case there’s no real reason checking both right?

File: cloudinary/apiCaller.js (hunk 1 above)
Comment (yaniv-li) [reply cloudinary/apiCaller.js:15]: removed

File: wtp/wtpResultsParser.js (hunk 3 above)
Comment (yaniv-li) [reply wtp/wtpResultsParser.js:66]: origImageList is the list before it gets filtered there is a chance that the LCP would get filtered out that is why I check to make sure it is there

File: cloudinary/apiCaller.js (hunk 2 above)
Comment (yaniv-li) [reply cloudinary/apiCaller.js:82]: I  think that will always return an LCP but I think it is more of a product question of what to do if there is no LCP

**Review by mikeys — APPROVED** [review]

**Pull request merged**